homeassistant/components/shelly/coordinator.py

```diff
@@ -145,6 +151,42 @@ def async_setup(self) -> None:
         )
         self.device_id = device_entry.id
 
+    async def _async_device_connect(self) -> None:
+        """Connect to a Shelly Block device."""
+        LOGGER.debug("Connecting to Shelly Device - %s", self.name)
+        try:
+            await self.device.initialize()
+            update_device_fw_info(self.hass, self.device, self.entry)
+        except DeviceConnectionError as err:
+            raise UpdateFailed(f"Device disconnected: {repr(err)}") from err
+        except InvalidAuthError:
+            self.entry.async_start_reauth(self.hass)
+        else:
```

Comment:
I personally prefer to add a `return` inside the `except InvalidAuthError` and it helps you save 1 indentation for the rest of the function.